src/tpool.rs

```diff
@@ -0,0 +1,54 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+pub use crate::bam::errors::{Error, Result};
+use crate::htslib;
+
+/// An HTSlib thread pool. Create a thread pool and use `set_thread_pool()` methods
+/// to share a thread pool across multiple BAM readers & writier.
+/// The Rust wrapper holds the htslib thread pool behind a Rc, and a Rc reference
+/// to the thread pool is held by each reader / writer so you don't need to
+/// explicitly manage the lifetime of the `ThreadPool`.
+#[derive(Clone, Debug)]
+pub struct ThreadPool {
+    pub(crate) handle: Rc<RefCell<InnerThreadPool>>,
```

Comment:
Should this be an Arc so you can share it across threads?

Reply:
We recently had https://github.com/rust-bio/rust-htslib/pull/234, would be interesting to hear your opinions on that one (related to `Arc` discussion(s)).